br/pkg/storage/storage.go

```diff
@@ -198,3 +205,29 @@ func New(ctx context.Context, backend *backuppb.StorageBackend, opts *ExternalSt
 		return nil, errors.Annotatef(berrors.ErrStorageInvalidConfig, "storage %T is not supported yet", backend)
 	}
 }
+
+// copy from `http.defaultTransportDialContext`
```

Comment:
another way is use go:linkname like https://github.com/pingcap/tidb/blob/5a305400a94464d226f95503c6a3068ef881e69d/br/pkg/lightning/backend/local/local_test.go#L752-L753